p2p/protocol/identify/id.go

```diff
@@ -317,6 +317,10 @@ func (ids *IDService) consumeReceivedPubKey(c network.Conn, kb []byte) {
 		return
 	}
 
+	if ids.Host.Peerstore().PubKey(rp) != nil {
+		return
+	}
```

Comment:
I think this will introduce problems. i believe that we need to verify the key matches here (not just use it), and recall having issues that led to this

Reply:
Things this code can currently do:

1. If `conn.RemotePeer()` returns `""` (insecure transport?) try to associate `""` with the new key (WTF?) which will fail (good) because `""` isn't a valid peer ID (phew).
2. If the peerstore doesn't have a key for `conn.RemotePeer()`, it adds it.
3. Log an error.

This change leaves 2 unchanged, 1 is just buggy (probably should have been `AddPubKey(np, newKey)`, and 3 shouldn't affect anything.